Replace BaseEventLoop with AbstractEventLoop

freundTech opened the pull request:
This is needed for the type annotations to work correctly with type-checkers.

asyncio.get_event_loop(), asyncio.get_running_loop() and similar functions all return AbstractEventLoop, leading to a type error if one tries to pass their return value to this library.

See https://github.com/python/typeshed/blob/master/stdlib/asyncio/events.pyi#L527

horazont commented:
@freundTech Hi! Thanks for the PR - I completely missed the notification though. 

Before we go further, could you please [include your PR description (as 72-char wide line-wrapped text) in the commit message](https://cbea.ms/git-commit/)? That would be most appreciated in order to have all necessary context for a change easily reachable from `git log`.

Thanks!

freundTech commented:
Done! I also replace the link to typeshed, as the line numbers had changed. I've now linked to a fixed revision.

horazont commented:
@freundTech Thanks! Now I can even trigger the pipeline run \o/. Could you maybe add the line wrapping to the commit message and leave blanks between the paragraphs?

freundTech commented:
I updated the commit message again and fixed the mypy and lint errors (mypy was complaining about an unused # type: ignore and flake8 about a line to long in inline documentation)

horazont commented:
Thanks a lot!
